### PR TITLE
feat(src/lib): adding a bunch of boilerplate sample stuff

### PR DESCRIPTION
**Pull request by mwflaher**

### CODE REVIEW

File: src/__tests__/index.test.tsx
Comment (mwflaher) [inline src/__tests__/index.test.tsx:1]: @cassiedoll would love some feedback here. is any of this insane? could we make any of it easier?

File: src/index.ts [hunk 1]
```diff
@@ -1,10 +1,35 @@
...
+import { EntityId } from 'mavenagi/api';
+
+import {
+  SAMPLE_ACTION_ID,
+  handleSampleTrigger,
+  installSampleAction,
+  refreshSampleKnowledge,
+  runSampleAction,
+  upsertSampleKnowledge,
+} from './lib';
```
Comment (sylvain-galineau) [inline src/index.ts:11]: With the path aliases I added in `tsconfig.json`, this can be `@lib`

File: src/index.ts [hunk 2]
```diff
@@ -34,11 +77,71 @@ export default {
...
+    console.info(`Handled sample trigger`, {
+      organizationId,
+      agentId,
+    });
+  },
+
+  /**
+   * Handler for any installed action(s)
+   */
+  async executeAction({
```
Comment (sylvain-galineau) [inline src/index.ts:131]: My initial thought with the `src/index.ts` vs the `lib` directory was that `src/index.ts` would only have `preInstall`+`postInstall`. Those would call `preInstallActions`/`postInstallActions` in `@lib/actions`, `preInstallKnowledge`/`postInstallKnowledge` in `@lib/knowledge` etc. And `executeAction` would also live in `@lib/actions` etc. 

I haven't done the `preInstall` and knowledge side but you can see this here for `postInstallActions` and `executeAction`

File: data/amendments.json [hunk 3]
```diff
@@ -0,0 +1,42 @@
+[
```
Comment (sylvain-galineau) [inline data/amendments.json:1]: "And then Matt made this "The package.json of America" t-shirt and retired on all that nerd money"

**Review by sylvain-galineau — APPROVED** [review]
LGTM with some suggestions on having the `lib` modules hold everything for a given capability: `preInstall`, `postInstall` and the corresponding handlers. `src/index.ts` would simply drive overall `preInstall`/`postInstall` and export everything?

**Pull request merged**